PackageInstallations_Troubleshoots.qmd

```diff
@@ -0,0 +1,125 @@
+---
+
+```{r}
+#| label = "setup",
+#| include = FALSE
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+
+First of all, please ensure that you have an up-to-date version of R
+(this note is based on R 4.3.1).
+
+## Mac M1 user
+
+When attempting to install miaverse packages, you may encounter
+installation failures related to the 'scuttle' and 'scatter'
+dependencies, which require a gcc compiler for installation. The error
+message might resemble the following:
+
+```
+ld: warning: directory not found for option
+'-L/opt/gfortran/lib/gcc/aarch64-apple-darwin20.0/12.2.0'
+ld: warning: directory not found for option '-L/opt/gfortran/lib'
+ld: library not found for -lgfortran
+clang: error: linker command failed with exit code 1 (use -v to see
+invocation)
+```
+
+This error indicates that R is unable to locate the 'libgfortran'
+during the package compilation process. If you encounter this error,
+you can use the following workaround:
+
+1. Install 2 essentials tools in order to compile R for macOS as
+described in [Mac-r-project](https://mac.r-project.org/tools/)
+
+- install the xcode: 
+	- sudo xcode-select --install
+- GNU Fortran compiler*
+	- download from
+	[here](https://mac.r-project.org/tools/gfortran-12.2-universal.pkg)
+	*you can also install from package manager for example homebrew,
+	but it is not reccomended.
+	
+2. Locate the gfortran using `which gfortran` to find the version of
+gfortran that is stored in your ~/.bashrc or ~/bash_profile. Make sure
+it pointing to the gfortran that you want to use. For example if you
+have both homebrew and mac-r project gfortran, you may want to add the
+mac-r project gfortran to your $PATH variable.
+    
+- add `export PATH="/opt/gfortran/bin:$PATH"` to your ~/.bash_profile
+or ~/.bashrc, followed by `source ~/.bash_profile` in your terminal
+
+3. Modify your `/Library/Frameworks/R.framework/Resources/etc/Makeconf`
+	
+	from: 
+	
+	```
+	FLIBS =  -L/opt/R/arm64/gfortran/lib/gcc/aarch64-apple-darwin20.2.0     /11.0.0 -L/opt/R/arm64/gfortran/lib -lgfortran -lemutls_w -lm
+	```
+	to*:
+	
+	```
+	FLIBS =  -L/opt/gfortran/lib/gcc/aarch64-apple-darwin20.0/12.2.0       -L/opt/gfortran/lib -lgfortran -lm -lquadmath
+	```
+*note that the path is depend on your installations site, sometimes it
+is located in ``/usr/local/gfortran/lib` so you have to change
+`/opt/gfortran/lib` accordingly in the lines above.
+
+After all of those steps, you can followed the installations based on
+Bioconductor page. For example for mia package:
+
+```
+if (!require("BiocManager", quietly = TRUE))
+    install.packages("BiocManager")
+
+BiocManager::install("mia")
+```
+
+If the installations prompt to compile from source, you can answer yes,
+and hopefully it can be succesfuly compiled. Good Luck!
+
+Follow-up reading:
+
+- https://support.bioconductor.org/p/9138121/ 
+- https://medium.com/biosyntax/following-up-library-dependency-when-compiling-r-packages-89f191b9f227
+
+
+Session info at the time of writing
```

Comment:
Yes indeed, but I am writing here to explain my env condition on the time of writing this troubleshoot. So it will not be updated anytime it needs to be rendered.